### PR TITLE
fix: support Xcode 26.4 BuildOperationMetrics JSON format change

### PR DESCRIPTION
**Pull request by giginet**
Closes #245

## Background

As of Xcode 26.4 Beta 2 (Build 17E5170d), the `BuildOperationMetrics` attachment JSON format in `.xcactivitylog` files has changed, causing XCLogParser to fail with:

```
Error: keyNotFound(CodingKeys(stringValue: "clangCacheHits", intValue: nil), ...)
```

### Investigation

We built the same sample project across Xcode 26.0 - 26.4 and compared the generated `.xcactivitylog` files:

| Xcode | SLF Version | BuildOperationMetrics Attachment |
|---|---|---|
| 26.0 | 11 | Not present |
| 26.1 | 11 | Not present |
| 26.2 | 12 | Not present |
| 26.3 | 12 | Not present |
| 26.4 | 12 | Present (new format) |

The old format (`clangCacheHits`, `clangCacheMisses`, `swiftCacheHits`, `swiftCacheMisses`) was used in Xcode 15.3 - 16.x. It was removed in Xcode 26.0 and reintroduced in Xcode 26.4 with a completely different schema.

**Old format (Xcode 15.3 - 16.x):**
```json
{"clangCacheHits":0,"clangCacheMisses":2,"swiftCacheHits":0,"swiftCacheMisses":8}
```

**New format (Xcode 26.4+):**
```json
{"counters":{},"taskCounters":{"SwiftDriver":{"moduleDependenciesNotValidatedTasks":1}}}
```

## Changes

- **`IDEActivityModel.swift`**: Make all `BuildOperationMetrics` fields optional and add new `counters: [String: Int]?` and `taskCounters: [String: [String: Int]]?` fields to support both old and new formats.
- **`ActivityParserTests.swift`**: Add `testParseIDEActivityLogSectionXcode264()` test with the new JSON format, verifying that new fields are parsed correctly and old fields are nil.
- **`docs/Xcactivitylog Format.md`**: Add SLF version history table, document `IDEActivityLogSectionAttachment` types and the `BuildOperationMetrics` format change.
- **`docs/JSON Format.md`**: Add Attachments section documenting `BuildOperationTaskMetrics`, `BuildOperationMetrics` (both formats), and `BuildOperationTaskBacktrace`.

### CODE REVIEW

**Comment by giginet** [comment]
The parsing crash on Xcode 26.4 logs has been resolved by #246. However, this PR is still needed to properly support the new columns (`counters` and `taskCounters`) introduced in the Xcode 26.4 BuildOperationMetrics format, rather than silently outputting dummy zero values.

**Comment by giginet** [comment]
@pepicrft @fortmarek Could you also check this?

File: Sources/XCLogParser/activityparser/IDEActivityModel.swift [hunk 1]
```diff
@@ -773,30 +773,42 @@ public class IDEActivityLogSectionAttachment: Encodable {
...
-            swiftCacheHits: Int = 0,
-            swiftCacheMisses: Int = 0
-        ) {
-            self.clangCacheHits = clangCacheHits
-            self.clangCacheMisses = clangCacheMisses
-            self.swiftCacheHits = swiftCacheHits
-            self.swiftCacheMisses = swiftCacheMisses
+    /// Build operation metrics whose JSON schema differs by Xcode version.
+    public enum BuildOperationMetrics: Encodable {
+        /// Xcode 15.3 - Xcode 16.x format
```
Comment (fortmarek) [inline Sources/XCLogParser/activityparser/IDEActivityModel.swift]:
```suggestion
        /// Xcode 15.3 - Xcode 26.3 format
```

File: Sources/XCLogParser/activityparser/IDEActivityModel.swift [hunk 2]
```diff
@@ -773,30 +773,42 @@ public class IDEActivityLogSectionAttachment: Encodable {
...
+        case v15_3(CacheMetrics)
+        /// Xcode 26.4+ format
+        case v26_4(CounterMetrics)
```
Comment (fortmarek) [inline Sources/XCLogParser/activityparser/IDEActivityModel.swift]: what do you think about translating `CacheMetrics` to `CounterMetrics` instead of forcing consumers to deal with both formats? In the end, `CounterMetrics` should also have clang/swift cache hits/misses?

**Review by pepicrft — DISMISSED** [review]

**Comment by giginet** [comment]
Verified that this branch correctly parses logs generated by the latest Xcode 26.4 RC (Build 17E192).

**Comment by giginet** [comment]
@fortmarek Sorry for my late response. I resolved the your points.
https://github.com/MobileNativeFoundation/XCLogParser/pull/244#discussion_r2891459566
https://github.com/MobileNativeFoundation/XCLogParser/pull/244/commits/460082cafd8a7bc5c3116d20331e43243a3e1fb4

Could you review this again?

**Review by fortmarek — DISMISSED** [review]
Thanks 🙏  should be good to merge once the CI passes.

**Review by pepicrft — APPROVED** [review]

**Comment by pepicrft** [comment]
ありがとう @giginet 🙏🏼
